lib/setupHooks/removeReferencesToRustToolchain.nix

```diff
@@ -0,0 +1,12 @@
+{
+  lib,
+  pkgs,
+  makeSetupHook,
+}:
+makeSetupHook { 
+  name = "removeReferencesToRustToolchain"; 
+  substitutions = {
+    storeDir = builtins.storeDir;
+  };
+  propagatedBuildInputs = [ pkgs.ripgrep ];
```

Comment:
Personally, I love ripgrep, but it would be nice if we can avoid the extra dependency here (most people will utilize the upstream cache, but in case they aren't or they are using a custom version of nixpkgs, it's one extra dependency to compile before their project can be run).

In this case it looks like ripgrep is being used to pull out the hash of the Rust toolchain, which we can also achieve via `grep` (which is already in the standard build environment)